### ISSUE TITLE
Some SQL filters don't get evaluated in a where linq chain.

### ISSUE THREAD

**Issue by ztone** [aspnet/EntityFramework#1645]
I might be missing something but I was just trying out EF7 and found something unexpected.

So this works
```c#
using (var ctx = new Context())
{
	var query = ctx.Data.Where(x => x.Date >= start && x.Date <= end && x.Excluded.Equals(true));
	var c = query.Count(); 
}
```
And get the expected SQl
```sql
exec sp_executesql N'SELECT COUNT(*)
FROM [Data] AS [x]
WHERE (([x].[Date] >= @p0 AND [x].[Date] <= @p0) AND [x].[Excluded] = @p1)',N'@p0 datetime,@p1 bit',@p0='2014-01-01 00:00:00',@p1=1
```

However chaining the filters ...
```c#
using (var ctx = new Context())
{
	var query = ctx.Data.Where(x => x.Date >= start && x.Date <= end);
	query = query.Where(x => x.Excluded.Equals(true));
	var c = query.Count(); 
}
```

... seems to overwrite the preceeding filters as only the last filter was evaluated in the SQL
```sql
exec sp_executesql N'SELECT COUNT(*)
FROM [Data] AS [x]
WHERE [x].[Excluded] = @p0',N'@p0 bit',@p0=1
```

The expression for reference
>{value(Microsoft.Data.Entity.Query.EntityQueryable`1[Company.DataAccess.Models.Data]).Where(x => ((x.Date >= Convert(value(Company.DataAccess.Services.Service+<>c__DisplayClass0).start)) AndAlso (x.Date <= Convert(value(Company.DataAccess.Services.Service+<>c__DisplayClass0).end)))).Where(x => x.Excluded.Equals(True))}

Using VS2015 Preview ASP 5 
dep : "EntityFramework": "7.0.0-beta1"

**Comment by rowanmiller** [comment]
Possibly already fixed

**Comment by maumar** [comment]
@ztone could you provide model used for this query as well (code for dbcontext + 'Data' entity). Thanks!

**Comment by ztone** [comment]
@maumar 

```c#
public class Context : DbContext
{
	public DbSet<Data> Data { get; set; }

	protected override void OnConfiguring(DbContextOptions options)
	{
		options.UseSqlServer(@"..");
	}

	protected override void OnModelCreating(ModelBuilder modelBuilder)
	{
		base.OnModelCreating(modelBuilder);
		modelBuilder.Entity<Data>().Key(sd => new { sd.IdType, sd.Id });
	}
}

public class Data
{
	public long IdType { get; set; }
	public byte Id { get; set; }
	public DateTime? Date { get; set; }
	public bool Excluded { get; set; }
}
```

**Comment by maumar** [comment]
This issue has been fixed in the current build, however scenario from the repro is currently blocked by https://github.com/aspnet/EntityFramework/pull/1740

**Comment by damccull** [comment]
Will this fix be available in nightlies?

**Comment by rowanmiller** [comment]
@damccull - The fix for this issue should already be there. The new issue that @maumar linked to is still active, but it will be available in nightlies once a fix is checked in (the turnaround is usually less than 24hrs from being committed to our dev branch to showing up in nightlies)

**Comment by maumar** [comment]
This should be checked in today.

**Comment by maumar** [comment]
Fixed in 2c4b376aef3d1faf5fa1963fe5253a2c18e31f4b
